### DISCUSSION TITLE
Adding Type Hinting

### DISCUSSION THREAD

**Discussion by glanham-jr** [searxng/searxng#2218] [Ideas]
I was directed here by unixfox, and I wanted to suggest the idea of adding type hinting to the project.

I'm wondering if we believe this would be a features worth implementing / maintaining. I find type hints make traversing code much easier to reason about, but runs the risk of becoming stale if not properly maintained.

**Comment by dalf** [comment]
I like typing, also `make test` starts pyright to warn about type issues.

However in many places it does not "fit" without change to the code. Example with this unfinished PR https://github.com/searxng/searxng/pull/1562 : `searx/results/models.py` contains the various result types. However, in `searx/webapp.py` you can see a `result = typing.cast(dict, result)` because this function here add and replace some fields. It makes typing difficult to declare without code change. This is one example, `searx/search/__init__.py` has some typing, it could be improved without too many issues.

Do you a module/place where you would like to start?

**Reply by glanham-jr** [reply]
No specific plan - I was thinking of first working on happy path sections of code that doesn't need much code change and can be quickly added. Then, if that looks good, maybe have a second branch for things that are tougher as you mentioned.

**Reply by dalf** [reply]
>  I was thinking of first working on happy path sections of code that doesn't need much code change and can be quickly added.

+1
